### ISSUE TITLE
Perhaps input is empty or misspecified

### ISSUE THREAD

**Issue by robertmckean** [google-research/google-research#1138]
I keep getting the following problem when trying to use the train function. I've tried everything I can think of.

INFO:tensorflow:Saving dict for global step 0: absolute_difference = 0.0, bass_spectrogram = 0.0, drums_spectrogram = 0.0, global_step = 0, loss = 0.0, other_spectrogram = 0.0, vocals_spectrogram = 0.0
INFO:tensorflow:Saving 'checkpoint_path' summary for global step 0: musdb_model\model.ckpt-0
WARNING:tensorflow:Training with estimator made no steps. Perhaps input is empty or misspecified.
INFO:tensorflow:Loss for final step: None.
INFO:spleeter:Model training done

I am running on Windows 10. I get the same issue with whatever version of Python/Tensorflow I use. This is my command line:

python -m spleeter train -p configs/musdb_config.json -d C:\Python\spleeter\configs --verbose

Is there anything anyone can think of for me to try?

I added extra print statements to try to track down. Here is the output:

We are in:  __main__
Entered entrypoint:  __main__
Someone called spleeter_callback:   __main__
Let us train:  __main__
INFO:spleeter:Just imported TF
Did this instantiate_pre?
We are in class InstrumentDatasetBuilder spleeter.dataset
We are in class class DatasetBuilder spleeter.dataset
Did this instantiate?
INFO:tensorflow:Using config: {'_model_dir': 'musdb_model', '_tf_random_seed': 3, '_save_summary_steps': 5, '_save_checkpoints_steps': 1000, '_save_checkpoints_secs': None, '_session_config': gpu_options {
  per_process_gpu_memory_fraction: 0.45
}
, '_keep_checkpoint_max': 2, '_keep_checkpoint_every_n_hours': 10000, '_log_step_count_steps': 10, '_train_distribute': None, '_device_fn': None, '_protocol': None, '_eval_distribute': None, '_experimental_distribute': None, '_experimental_max_worker_delay_secs': None, '_session_creation_timeout_secs': 7200, '_checkpoint_save_graph_def': True, '_service': None, '_cluster_spec': ClusterSpec({}), '_task_type': 'worker', '_task_id': 0, '_global_id_in_cluster': 0, '_master': '', '_evaluation_master': '', '_is_chief': True, '_num_ps_replicas': 0, '_num_worker_replicas': 1}
Ready to jump?......
Just jumped?......
INFO:spleeter:Start model training
estimator........:  <tensorflow_estimator.python.estimator.estimator.EstimatorV2 object at 0x0000028688343E50>
train_spec.......:  TrainSpec(input_fn=functools.partial(<function get_training_dataset at 0x00000286ABE050D0>, {'train_csv': 'configs/musdb_train.csv', 'validation_csv': 'configs/musdb_validation.csv', 'model_dir': 'musdb_model', 'mix_name': 'mix', 'instrument_list': ['vocals', 'drums', 'bass', 'other'], 'sample_rate': 44100, 'frame_length': 4096, 'frame_step': 1024, 'T': 512, 'F': 1024, 'n_channels': 2, 'n_chunks_per_song': 40, 'separation_exponent': 2, 'mask_extension': 'zeros', 'learning_rate': 0.0001, 'batch_size': 4, 'training_cache': 'cache/training', 'validation_cache': 'cache/validation', 'train_max_steps': 200000, 'throttle_secs': 1800, 'random_seed': 3, 'save_checkpoints_steps': 1000, 'save_summary_steps': 5, 'model': {'type': 'unet.unet', 'params': {'conv_activation': 'ELU', 'deconv_activation': 'ELU'}}}, <spleeter.audio.ffmpeg.FFMPEGProcessAudioAdapter object at 0x0000028688343DC0>, 'C:\\Python\\spleeter\\configs'), max_steps=200000, hooks=(), saving_listeners=())
evaluation_spec..:  EvalSpec(input_fn=functools.partial(<function get_validation_dataset at 0x00000286AD9AA9D0>, {'train_csv': 'configs/musdb_train.csv', 'validation_csv': 'configs/musdb_validation.csv', 'model_dir': 'musdb_model', 'mix_name': 'mix', 'instrument_list': ['vocals', 'drums', 'bass', 'other'], 'sample_rate': 44100, 'frame_length': 4096, 'frame_step': 1024, 'T': 512, 'F': 1024, 'n_channels': 2, 'n_chunks_per_song': 40, 'separation_exponent': 2, 'mask_extension': 'zeros', 'learning_rate': 0.0001, 'batch_size': 4, 'training_cache': 'cache/training', 'validation_cache': 'cache/validation', 'train_max_steps': 200000, 'throttle_secs': 1800, 'random_seed': 3, 'save_checkpoints_steps': 1000, 'save_summary_steps': 5, 'model': {'type': 'unet.unet', 'params': {'conv_activation': 'ELU', 'deconv_activation': 'ELU'}}}, <spleeter.audio.ffmpeg.FFMPEGProcessAudioAdapter object at 0x0000028688343DC0>, 'C:\\Python\\spleeter\\configs'), steps=None, name=None, hooks=(), exporters=(), start_delay_secs=120, throttle_secs=1800)

INFO:tensorflow:Not using Distribute Coordinator.
INFO:tensorflow:Running training and evaluation locally (non-distributed).
INFO:tensorflow:Start train and evaluate loop. The evaluate will happen after every checkpoint. 
        Checkpoint frequency is determined based on RunConfig arguments: 
            save_checkpoints_steps 1000 or save_checkpoints_secs None.
WARNING:tensorflow:From C:\Users\windo\anaconda3\envs\g397\lib\site-packages\tensorflow\python\training\training_util.py:
                235: Variable.initialized_value (from tensorflow.python.ops.variables) is deprecated and will be 
                    removed in a future version.
Instructions for updating:
Use Variable.read_value. 
Variables in 2.X are initialized automatically both in eager and graph (inside tf.defun) contexts.

We are in get_training_dataset:  spleeter.dataset
Loaded from:  C:\Python\spleeter\spleeter\dataset.py
Here is df:  
                                              mix_path  ...    duration
0    train\A Classic Education - NightOwl\mixture.wav  ...  171.247166
1                 train\ANiMAL - Clinic A\mixture.wav  ...  237.865215
2               train\ANiMAL - Easy Tiger\mixture.wav  ...  205.473379
3           train\Actions - Devil's Words\mixture.wav  ...  196.626576
4      train\Actions - South Of The Water\mixture.wav  ...  176.610975
..                                                ...  ...         ...
81                train\Triviul - Dorothy\mixture.wav  ...  187.361814
82  train\Voelund - Comfort Lives In Belief\mixtur...  ...  209.908390
83            train\Wall Of Death - Femme\mixture.wav  ...  238.933333
84     train\Young Griffo - Blood To Bone\mixture.wav  ...  254.397823
85            train\Young Griffo - Facade\mixture.wav  ...  167.857052

[86 rows x 6 columns]
dataset....................... <TensorSliceDataset shapes: {mix_path: (), vocals_path: (), drums_path: (), bass_path: (), other_path: (), duration: ()}, types: {mix_path: tf.string, vocals_path: tf.string, drums_path: tf.string, bass_path: tf.string, other_path: tf.string, duration: tf.float64}>
Loaded from:  C:\Python\spleeter\spleeter\utils\tensor.py
dataset:   <TensorSliceDataset shapes: {mix_path: (), vocals_path: (), drums_path: (), bass_path: (), other_path: (), duration: ()}, types: {mix_path: tf.string, vocals_path: tf.string, drums_path: tf.string, bass_path: tf.string, other_path: tf.string, duration: tf.float64}>
dataset before:  <ShuffleDataset shapes: {mix_path: (), vocals_path: (), drums_path: (), bass_path: (), other_path: (), duration: (), start: ()}, types: {mix_path: tf.string, vocals_path: tf.string, drums_path: tf.string, bass_path: tf.string, other_path: tf.string, duration: tf.float64, start: tf.float64}>
We are in load_waveform:  spleeter.dataset
We are in compute_spectrogram:  spleeter.dataset
We are in compute_spectrogram_tf in:  spleeter.audio.spectrogram
here is the tensor:   Tensor("transpose_1:0", shape=(None, None, None), dtype=complex64)
We are in load_waveform:  spleeter.dataset
We are in compute_spectrogram:  spleeter.dataset
We are in compute_spectrogram_tf in:  spleeter.audio.spectrogram
here is the tensor:   Tensor("transpose_1:0", shape=(None, None, None), dtype=complex64)
We are in load_waveform:  spleeter.dataset
We are in compute_spectrogram:  spleeter.dataset
We are in compute_spectrogram_tf in:  spleeter.audio.spectrogram
here is the tensor:   Tensor("transpose_1:0", shape=(None, None, None), dtype=complex64)
We are in load_waveform:  spleeter.dataset
We are in compute_spectrogram:  spleeter.dataset
We are in compute_spectrogram_tf in:  spleeter.audio.spectrogram
here is the tensor:   Tensor("transpose_1:0", shape=(None, None, None), dtype=complex64)
We are in load_waveform:  spleeter.dataset
We are in compute_spectrogram:  spleeter.dataset
We are in compute_spectrogram_tf in:  spleeter.audio.spectrogram
here is the tensor:   Tensor("transpose_1:0", shape=(None, None, None), dtype=complex64)
INFO:tensorflow:Calling model_fn.
We are in:  spleeter.model
We are in:  spleeter.model
We are in:  spleeter.model
We are in:  spleeter.model
We are in:  spleeter.model
We are in:  spleeter.model.functions.unet
We are in apply_unet:  spleeter.model.functions.unet
INFO:tensorflow:Apply unet for vocals_spectrogram
We are in _get_conv_activation_layer spleeter.model.functions.unet
We are in _get_deconv_activation_layer:  spleeter.model.functions.unet
WARNING:tensorflow:From C:\Users\windo\anaconda3\envs\g397\lib\site-packages\tensorflow\python\keras\layers\normalization.py:534: _colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
We are in apply_unet:  spleeter.model.functions.unet
INFO:tensorflow:Apply unet for drums_spectrogram
We are in _get_conv_activation_layer spleeter.model.functions.unet
We are in _get_deconv_activation_layer:  spleeter.model.functions.unet
We are in apply_unet:  spleeter.model.functions.unet
INFO:tensorflow:Apply unet for bass_spectrogram
We are in _get_conv_activation_layer spleeter.model.functions.unet
We are in _get_deconv_activation_layer:  spleeter.model.functions.unet
We are in apply_unet:  spleeter.model.functions.unet
INFO:tensorflow:Apply unet for other_spectrogram
We are in _get_conv_activation_layer spleeter.model.functions.unet
We are in _get_deconv_activation_layer:  spleeter.model.functions.unet
We are in:  spleeter.model
INFO:tensorflow:Done calling model_fn.
INFO:tensorflow:Create CheckpointSaverHook.
INFO:tensorflow:Graph was finalized.
INFO:tensorflow:Restoring parameters from musdb_model\model.ckpt-0
WARNING:tensorflow:From C:\Users\windo\anaconda3\envs\g397\lib\site-packages\tensorflow\python\training\saver.py:1078: get_checkpoint_mtimes (from tensorflow.python.training.checkpoint_management) is deprecated and will be removed in a future version.
Instructions for updating:
Use standard file utilities to get mtimes.
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Calling checkpoint listeners before saving checkpoint 0...
INFO:tensorflow:Saving checkpoints for 0 into musdb_model\model.ckpt.
INFO:tensorflow:Calling checkpoint listeners after saving checkpoint 0...
We are in get_validation_dataset:  spleeter.dataset
Loaded from:  C:\Python\spleeter\spleeter\dataset.py
Here is df:  
                                              mix_path  ...    duration
0                 train\ANiMAL - Rockshow\mixture.wav  ...  165.511837
1        train\Actions - One Minute Smile\mixture.wav  ...  163.375601
2   train\Alexander Ross - Goodbye Bolero\mixture.wav  ...  418.632562
3   train\Clara Berry And Wooldog - Waltz For My V...  ...  175.240998
4        train\Fergessen - Nos Palpitants\mixture.wav  ...  198.228753
5           train\James May - On The Line\mixture.wav  ...  256.092880
6    train\Johnny Lokke - Promises & Lies\mixture.wav  ...  285.814422
7                train\Leaf - Summerghost\mixture.wav  ...  231.804807
8              train\Meaxic - Take A Step\mixture.wav  ...  282.517188
9   train\Patrick Talbot - A Reason To Leave\mixtu...  ...  259.552653
10        train\Skelpolu - Human Mistakes\mixture.wav  ...  324.498866
11      train\Traffic Experiment - Sirens\mixture.wav  ...  421.279637
12             train\Triviul - Angelsaint\mixture.wav  ...  236.704218
13           train\Young Griffo - Pennies\mixture.wav  ...  277.803537

[14 rows x 6 columns]
dataset....................... <TensorSliceDataset shapes: {mix_path: (), vocals_path: (), drums_path: (), bass_path: (), other_path: (), duration: ()}, types: {mix_path: tf.string, vocals_path: tf.string, drums_path: tf.string, bass_path: tf.string, other_path: tf.string, duration: tf.float64}>
Loaded from:  C:\Python\spleeter\spleeter\utils\tensor.py
dataset:   <TensorSliceDataset shapes: {mix_path: (), vocals_path: (), drums_path: (), bass_path: (), other_path: (), duration: ()}, types: {mix_path: tf.string, vocals_path: tf.string, drums_path: tf.string, bass_path: tf.string, other_path: tf.string, duration: tf.float64}>
dataset before:  <MapDataset shapes: {mix_path: (), vocals_path: (), drums_path: (), bass_path: (), other_path: (), duration: (), start: ()}, types: {mix_path: tf.string, vocals_path: tf.string, drums_path: tf.string, bass_path: tf.string, other_path: tf.string, duration: tf.float64, start: tf.float64}>
We are in load_waveform:  spleeter.dataset
We are in compute_spectrogram:  spleeter.dataset
We are in compute_spectrogram_tf in:  spleeter.audio.spectrogram
here is the tensor:   Tensor("transpose_1:0", shape=(None, None, None), dtype=complex64)
We are in load_waveform:  spleeter.dataset
We are in compute_spectrogram:  spleeter.dataset
We are in compute_spectrogram_tf in:  spleeter.audio.spectrogram
here is the tensor:   Tensor("transpose_1:0", shape=(None, None, None), dtype=complex64)
We are in load_waveform:  spleeter.dataset
We are in compute_spectrogram:  spleeter.dataset
We are in compute_spectrogram_tf in:  spleeter.audio.spectrogram
here is the tensor:   Tensor("transpose_1:0", shape=(None, None, None), dtype=complex64)
We are in load_waveform:  spleeter.dataset
We are in compute_spectrogram:  spleeter.dataset
We are in compute_spectrogram_tf in:  spleeter.audio.spectrogram
here is the tensor:   Tensor("transpose_1:0", shape=(None, None, None), dtype=complex64)
We are in load_waveform:  spleeter.dataset
We are in compute_spectrogram:  spleeter.dataset
We are in compute_spectrogram_tf in:  spleeter.audio.spectrogram
here is the tensor:   Tensor("transpose_1:0", shape=(None, None, None), dtype=complex64)
INFO:tensorflow:Calling model_fn.
We are in:  spleeter.model
We are in:  spleeter.model
We are in:  spleeter.model
We are in:  spleeter.model
We are in:  spleeter.model
We are in:  spleeter.model.functions.unet
We are in apply_unet:  spleeter.model.functions.unet
INFO:tensorflow:Apply unet for vocals_spectrogram
We are in _get_conv_activation_layer spleeter.model.functions.unet
We are in _get_deconv_activation_layer:  spleeter.model.functions.unet
We are in apply_unet:  spleeter.model.functions.unet
INFO:tensorflow:Apply unet for drums_spectrogram
We are in _get_conv_activation_layer spleeter.model.functions.unet
We are in _get_deconv_activation_layer:  spleeter.model.functions.unet
We are in apply_unet:  spleeter.model.functions.unet
INFO:tensorflow:Apply unet for bass_spectrogram
We are in _get_conv_activation_layer spleeter.model.functions.unet
We are in _get_deconv_activation_layer:  spleeter.model.functions.unet
We are in apply_unet:  spleeter.model.functions.unet
INFO:tensorflow:Apply unet for other_spectrogram
We are in _get_conv_activation_layer spleeter.model.functions.unet
We are in _get_deconv_activation_layer:  spleeter.model.functions.unet
INFO:tensorflow:Done calling model_fn.
INFO:tensorflow:Starting evaluation at 2022-05-29T18:24:50
INFO:tensorflow:Graph was finalized.
INFO:tensorflow:Restoring parameters from musdb_model\model.ckpt-0
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Inference Time : 4.72633s
INFO:tensorflow:Finished evaluation at 2022-05-29-18:24:54
INFO:tensorflow:Saving dict for global step 0: absolute_difference = 0.0, bass_spectrogram = 0.0, drums_spectrogram = 0.0, global_step = 0, loss = 0.0, other_spectrogram = 0.0, vocals_spectrogram = 0.0
INFO:tensorflow:Saving 'checkpoint_path' summary for global step 0: musdb_model\model.ckpt-0
WARNING:tensorflow:Training with estimator made no steps. Perhaps input is empty or misspecified.
INFO:tensorflow:Loss for final step: None.
INFO:spleeter:Model training done
